doc/libpmem2/pmem2_config_new.3.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: manual
+Content-Style: 'text/css'
+title: _MP(PMEM2_CONFIG_NEW, 7)
+collection: libpmem2
+header: PMDK
+date: pmem2 API version 1.0
+...
+
+[comment]: <> (Copyright 2019, Intel Corporation)
+
+[comment]: <> (Redistribution and use in source and binary forms, with or without)
+[comment]: <> (modification, are permitted provided that the following conditions)
+[comment]: <> (are met:)
+[comment]: <> (    * Redistributions of source code must retain the above copyright)
+[comment]: <> (      notice, this list of conditions and the following disclaimer.)
+[comment]: <> (    * Redistributions in binary form must reproduce the above copyright)
+[comment]: <> (      notice, this list of conditions and the following disclaimer in)
+[comment]: <> (      the documentation and/or other materials provided with the)
+[comment]: <> (      distribution.)
+[comment]: <> (    * Neither the name of the copyright holder nor the names of its)
+[comment]: <> (      contributors may be used to endorse or promote products derived)
+[comment]: <> (      from this software without specific prior written permission.)
+
+[comment]: <> (THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS)
+[comment]: <> ("AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT)
+[comment]: <> (LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR)
+[comment]: <> (A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT)
+[comment]: <> (OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,)
+[comment]: <> (SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT)
+[comment]: <> (LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,)
+[comment]: <> (DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY)
+[comment]: <> (THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT)
+[comment]: <> ((INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE)
+[comment]: <> (OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.)
+
+[comment]: <> (pmem2_config_new.3 -- man page for pmem2_config_new and pmem2_config_delete
+
+[NAME](#name)<br />
+[SYNOPSIS](#synopsis)<br />
+[DESCRIPTION](#description)<br />
+[RETURN VALUE](#return-value)<br />
+[SEE ALSO](#see-also)<br />
+
+# NAME #
+
+**pmem2_config_new**(), **pmem2_config_delete**() - allocate and free a 
+configuration for a libpmem2 mapping
+
+# SYNOPSIS #
+
+```c
+#include <libpmem2.h>
+
+struct pmem2_config;
+int pmem2_config_new(struct pmem2_config **cfg);
+int pmem2_config_delete(struct pmem2_config **cfg);
+```
+
+# DESCRIPTION #
+
+The **pmem2_config_new**() function allocates *pmem2_config*, and places it in *\*cfg*. 
+Returned structure is initialized to default parameters.
```

Comment:
this depends on each parameter, do you really want to document each one here?

Reply:
That's not what I'm saying.
As is, the sentence doesn't add anything, and might as well be removed. You could say that:
`New configuration is always initialized with default values for all possible parameters, which are specified alongside the corresponding setter function.`

But maybe we should list all parameters somewhere? So that you don't have to look for them in all the various separate man pages that we plan on having?